.gitlab-ci.yml

```diff
@@ -0,0 +1,58 @@
+variables:
+  AWS_DEFAULT_REGION: "us-east-1"
+  DSS_SECRETS_STORE: "dcp/dss"
+  DSS_TEST_STAGE: "dev"
+  BOTO_CONFIG: "/dev/null"
+
+stages:
+    - test
+
+.linux-test-base:
+  stage: test
+  image: ubuntu:18.04
+  before_script:
+    - IFS='.'; semver=($PYTHON_VERSION); PYTHON_MAJOR=${semver[0]}; unset IFS
+    - export DEBIAN_FRONTEND=noninteractive
+    - apt-get update --quiet
+    - apt-get install --assume-yes --no-install-recommends software-properties-common
+    - add-apt-repository --yes ppa:deadsnakes/ppa
+    - apt-get update --quiet
+    - apt-get install --assume-yes --no-install-recommends curl jq make python${PYTHON_MAJOR/2/}-pip python${PYTHON_VERSION}-dev
+    - python${PYTHON_VERSION} -m pip install --upgrade pip==10.0.1
+    - python${PYTHON_VERSION} -m pip install virtualenv==16.0.0
+    - rm -f /usr/bin/python; ln -s /usr/bin/python${PYTHON_VERSION} /usr/bin/python
+    - rm -f /usr/bin/pip; ln -s /usr/bin/pip${PYTHON_MAJOR} /usr/bin/pip
+    - virtualenv ~/venv
+    - source ~/venv/bin/activate
+    - pip install -r requirements-dev.txt
+    - scripts/fetch_secret.sh gcp-credentials.json > gcp-credentials.json
+    - export GOOGLE_APPLICATION_CREDENTIALS=$(pwd -P)/gcp-credentials.json
+    - sed -i -e s/dss.data.humancellatlas.org/dss.$DSS_TEST_STAGE.data.humancellatlas.org/ hca/default_config.json
+    - make install
+  script:
+    - make test
+  except:
+    - tags
+
+after_script:
+    - bash <(curl -s https://codecov.io/bash)
```

Comment:
Hi @xbrianh, I suspect this `after_script` section for code coverage can be removed, do you happen to know if that is true? In short, it appears the necessary code coverage instrumentation is in the Makefile.
Here are observations regarding why it is currently here, and why I suspect it can be removed:
* Looking at the `data-store` repo:
    * The `.travis.yml` has an entry like the above, yet the `.gitlab-ci.yml` does not.
    * The Makefile includes code coverage instrumentation and report consolidation
    * The GitHub PR checks include the codecov results
* Looking at the `dcp-cli`:
   * The `.travis.yml` has an entry like the above, the `.gitlab-ci.yml` currently does, yet perhaps *should not*
    * The Makefile includes code coverage instrumentation and report consolidation
    * The GitHub PR checks include the codecov results
So, is there any reason to keep the `after_script` section above?
I can try taking it out and checking for any untoward effects, but I'm not sure what all the potential undesired effects might be, so I thought I'd ask. :)

Reply:
Your analysis indicates that we can drop `after_script` and expect code coverage to keep working as expected. I'm not aware of any possible negative affects, so I think we should give it a shot.

Reply:
Here's a description of what this does:

[Codecov Bash Uploader](https://docs.codecov.io/docs/about-the-codecov-bash-uploader)

Given that the PR code coverage reports are accessed at `https://codecov.io/...` this may be important, unless GitLab is providing the coverage reports some other way.
GitLab can publish coverage reports as described here:

[Publish code coverage report with GitLab Pages](https://about.gitlab.com/2016/11/03/publish-code-coverage-report-with-gitlab-pages/)

At present, it seems the code coverage reports on our GitHub PRs are coming from the Travis builds, which are running the CodeCove bash uploader, and are not coming from our GitLab builds. *If* that is the intended behavior, now and going forward, then I we we wouldn't need to, and shouldn't, run the Codecov Bash uploader for our GitLab builds. 
Correct? Comments?